Won't build on node 12

Shifted to Ubuntu 20.04, so thought I'd get myself up to date with Node 12 (current LTS). Then remembered I had this problem before - node-osmium won't build on Node 12 because v8 has [deprecated lots of non-maybe APIs](https://github.com/joyeecheung/node/blob/v8-maybe-doc/CPP_STYLE_GUIDE.md#use-maybe-version-of-v8-apis).

```
mat@devbox:~/dev$ git clone https://github.com/osmcode/node-osmium.git
Cloning into 'node-osmium'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 2548 (delta 4), reused 7 (delta 2), pack-reused 2534
Receiving objects: 100% (2548/2548), 604.51 KiB | 1.42 MiB/s, done.
Resolving deltas: 100% (1806/1806), done.
mat@devbox:~/dev$ cd node-osmium
mat@devbox:~/dev/node-osmium$ make
npm install node-pre-gyp
npm notice created a lockfile as package-lock.json. You should commit this file.
+ node-pre-gyp@0.10.3
added 66 packages from 25 contributors and audited 66 packages in 4.607s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

npm install `node -e "console.log(Object.keys(require('./package.json').dependencies).join(' '))"` \
`node -e "console.log(Object.keys(require('./package.json').devDependencies).join(' '))"` --clang=1
npm WARN deprecated turf-difference@2.0.0: Turf packages are now namespaced: please use @turf/difference instead
npm WARN deprecated geojson-normalize@0.0.0: This module is now under the @mapbox namespace: install @mapbox/geojson-normalize instead
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated mkdirp@0.3.0: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
+ aws-sdk@2.720.0
+ nan@2.14.1
+ geojson-coords@0.0.0
+ turf-difference@2.0.0
+ node-pre-gyp@0.10.3
+ mocha@2.5.3
+ wellknown@0.4.2
+ osm-testdata@1.0.0
added 59 packages from 372 contributors, updated 8 packages and audited 125 packages in 9.863s

1 package is looking for funding
  run `npm fund` for details

found 9 vulnerabilities (7 low, 1 high, 1 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
./node_modules/.bin/node-pre-gyp configure --loglevel=error --clang=1
./node_modules/.bin/node-pre-gyp build --loglevel=error --clang=1
make[1]: Entering directory '/home/mat/dev/node-osmium/build'
  CXX(target) Release/obj.target/osmium/src/apply.o
In file included from ../src/apply.cpp:6:
../node_modules/nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../node_modules/nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
 2294 |     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
      |                                                              ^
In file included from ../src/apply.cpp:22:
../src/utils.hpp: In function ‘v8::Local<v8::Object> node_osmium::new_external(Args&& ...)’:
../src/utils.hpp:38:13: error: ‘Handle’ is not a member of ‘v8’
   38 |         v8::Handle<v8::Value> ext = Nan::New<v8::External>(new T(std::forward<Args>(args)...));
      |             ^~~~~~
../src/utils.hpp:38:29: error: expected primary-expression before ‘>’ token
   38 |         v8::Handle<v8::Value> ext = Nan::New<v8::External>(new T(std::forward<Args>(args)...));
      |                             ^
../src/utils.hpp:38:31: error: ‘ext’ was not declared in this scope; did you mean ‘exit’?
   38 |         v8::Handle<v8::Value> ext = Nan::New<v8::External>(new T(std::forward<Args>(args)...));
      |                               ^~~
      |                               exit
../src/utils.hpp:41:68: error: no matching function for call to ‘v8::Object::ToObject()’
   41 |         return scope.Escape(maybe_local.ToLocalChecked()->ToObject());
      |                                                                    ^
In file included from /home/mat/.cache/node-gyp/12.18.3/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/apply.cpp:6:
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2672:44: note: candidate: ‘v8::MaybeLocal<v8::Object> v8::Value::ToObject(v8::Local<v8::Context>) const’
 2672 |   V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
      |                                            ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2672:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/mat/.cache/node-gyp/12.18.3/include/node/v8-internal.h:14,
                 from /home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:27,
                 from /home/mat/.cache/node-gyp/12.18.3/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/apply.cpp:6:
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2686:31: note: candidate: ‘v8::Local<v8::Object> v8::Value::ToObject(v8::Isolate*) const’
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2686:31: note:   candidate expects 1 argument, 0 provided
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
../src/apply.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE node_osmium::apply(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/apply.cpp:164:46: error: no matching function for call to ‘v8::Value::ToObject()’
  164 |                 auto obj = info[i]->ToObject();
      |                                              ^
In file included from /home/mat/.cache/node-gyp/12.18.3/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/apply.cpp:6:
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2672:44: note: candidate: ‘v8::MaybeLocal<v8::Object> v8::Value::ToObject(v8::Local<v8::Context>) const’
 2672 |   V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
      |                                            ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2672:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/mat/.cache/node-gyp/12.18.3/include/node/v8-internal.h:14,
                 from /home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:27,
                 from /home/mat/.cache/node-gyp/12.18.3/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/apply.cpp:6:
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2686:31: note: candidate: ‘v8::Local<v8::Object> v8::Value::ToObject(v8::Isolate*) const’
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2686:31: note:   candidate expects 1 argument, 0 provided
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
../src/apply.cpp:175:49: error: no matching function for call to ‘v8::Value::ToObject()’
  175 |                 auto source = info[0]->ToObject();
      |                                                 ^
In file included from /home/mat/.cache/node-gyp/12.18.3/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/apply.cpp:6:
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2672:44: note: candidate: ‘v8::MaybeLocal<v8::Object> v8::Value::ToObject(v8::Local<v8::Context>) const’
 2672 |   V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
      |                                            ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2672:44: note:   candidate expects 1 argument, 0 provided
In file included from /home/mat/.cache/node-gyp/12.18.3/include/node/v8-internal.h:14,
                 from /home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:27,
                 from /home/mat/.cache/node-gyp/12.18.3/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/apply.cpp:6:
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2686:31: note: candidate: ‘v8::Local<v8::Object> v8::Value::ToObject(v8::Isolate*) const’
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8.h:2686:31: note:   candidate expects 1 argument, 0 provided
 2686 |                 Local<Object> ToObject(Isolate* isolate) const);
      |                               ^~~~~~~~
/home/mat/.cache/node-gyp/12.18.3/include/node/v8config.h:328:3: note: in definition of macro ‘V8_DEPRECATED’
  328 |   declarator __attribute__((deprecated(message)))
      |   ^~~~~~~~~~
make[1]: *** [osmium.target.mk:137: Release/obj.target/osmium/src/apply.o] Error 1
make[1]: Leaving directory '/home/mat/dev/node-osmium/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 5.4.0-42-generic
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--loglevel=error" "--clang=1" "--module=/home/mat/dev/node-osmium/lib/binding/osmium.node" "--module_name=osmium" "--module_path=/home/mat/dev/node-osmium/lib/binding" "--napi_version=6" "--node_abi_napi=napi"
gyp ERR! cwd /home/mat/dev/node-osmium
gyp ERR! node -v v12.18.3
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --loglevel=error --clang=1 --module=/home/mat/dev/node-osmium/lib/binding/osmium.node --module_name=osmium --module_path=/home/mat/dev/node-osmium/lib/binding --napi_version=6 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/mat/dev/node-osmium/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
node-pre-gyp ERR! System Linux 5.4.0-42-generic
node-pre-gyp ERR! command "/usr/bin/node" "/home/mat/dev/node-osmium/node_modules/.bin/node-pre-gyp" "build" "--loglevel=error" "--clang=1"
node-pre-gyp ERR! cwd /home/mat/dev/node-osmium
node-pre-gyp ERR! node -v v12.18.3
node-pre-gyp ERR! node-pre-gyp -v v0.10.3
node-pre-gyp ERR! not ok
Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --loglevel=error --clang=1 --module=/home/mat/dev/node-osmium/lib/binding/osmium.node --module_name=osmium --module_path=/home/mat/dev/node-osmium/lib/binding --napi_version=6 --node_abi_napi=napi' (1)
make: *** [Makefile:31: build-all] Error 1
mat@devbox:~/dev/node-osmium$
```

Appreciate there are no binaries for Node 12 (#108), but it would be nice if it compiled.

I had a good stab at trying to fix this in d8eb02c475a8aad2c0a7c50748d6123b12ca6887 (no guarantees any of it works - I was trying to fix the deprecation errors before trying to actually get it to work) but got stuck with [utils.hpp](https://github.com/osmcode/node-osmium/blob/master/src/utils.hpp#L35-L43):

```cpp
    template<class T, class... Args>
    v8::Local<v8::Object> new_external(Args&&... args) {
        Nan::EscapableHandleScope scope;
        v8::Handle<v8::Value> ext = Nan::New<v8::External>(new T(std::forward<Args>(args)...));
        Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(Nan::New(T::constructor)->GetFunction(), 1, &ext);
        if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Buffer instance");
        return scope.Escape(maybe_local.ToLocalChecked()->ToObject());

    }
```

```
../src/utils.hpp:39:67: error: no matching function for call to ‘v8::FunctionTemplate::GetFunction()’
   39 |         Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(Nan::New(T::constructor)->GetFunction(), 1, &ext);
      |                                                   ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Where might I get the context for `GetFunction` from?

Frustratingly can't get rid of these repeated warnings, but nodejs/nan#878 suggests they're safe to ignore:

```
  CXX(target) Release/obj.target/osmium/src/apply.o
In file included from ../src/apply.cpp:6:
../node_modules/nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../node_modules/nan/nan.h:2294:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
 2294 |     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
      |                                                              ^
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -45,6 +45,12 @@ matrix:
      - os: linux
        compiler: clang
        env: NODE_VERSION="10"
+     - os: linux
+       compiler: clang
+       env: NODE_VERSION="12"
+     - os: linux
+       compiler: clang
+       env: NODE_VERSION="14"
      # OS X
      - os: osx
        osx_image: xcode8
@@ -62,6 +68,14 @@ matrix:
        osx_image: xcode8
        compiler: clang
        env: NODE_VERSION="10"
+     - os: osx
+       osx_image: xcode8
+       compiler: clang
+       env: NODE_VERSION="12"
+     - os: osx
+       osx_image: xcode8
+       compiler: clang
+       env: NODE_VERSION="14"
 
 env:
   global:

--- a/src/apply.cpp
+++ b/src/apply.cpp
@@ -161,7 +161,7 @@ namespace node_osmium {
                     Nan::ThrowTypeError(Nan::New("please provide handler objects as second and further parameters to apply()").ToLocalChecked());
                     return;
                 }
-                auto obj = info[i]->ToObject();
+                auto obj = info[i]->ToObject(info.GetIsolate()->GetCurrentContext()).ToLocalChecked();
                 if (Nan::New(JSHandler::constructor)->HasInstance(obj)) {
                     handlers.push_back(unwrap<JSHandler>(obj));
                 } else if (Nan::New(LocationHandlerWrap::constructor)->HasInstance(obj)) {
@@ -172,7 +172,7 @@ namespace node_osmium {
             }
 
             try {
-                auto source = info[0]->ToObject();
+                auto source = info[0]->ToObject(info.GetIsolate()->GetCurrentContext()).ToLocalChecked();
                 if (Nan::New(BasicReaderWrap::constructor)->HasInstance(source)) {
                     osmium::io::Reader& reader = unwrap<BasicReaderWrap>(source);
 

--- a/src/basic_reader_wrap.cpp
+++ b/src/basic_reader_wrap.cpp
@@ -32,7 +32,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "close", close);
         Nan::SetPrototypeMethod(lcons, "read", read);
         Nan::SetPrototypeMethod(lcons, "read_all", read_all);
-        target->Set(Nan::New(symbol_BasicReader), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_BasicReader), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -52,7 +52,7 @@ namespace node_osmium {
                     Nan::ThrowTypeError(Nan::New("Second argument to BasicReader constructor must be object").ToLocalChecked());
                     return;
                 }
-                read_which_entities = object_to_entity_bits(info[1]->ToObject());
+                read_which_entities = object_to_entity_bits(info[1]->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
             }
             if (info[0]->IsString()) {
                 Nan::Utf8String filename { info[0] };
@@ -61,8 +61,8 @@ namespace node_osmium {
                 reader_wrap->Wrap(info.This());
                 info.GetReturnValue().Set(info.This());
                 return;
-            } else if (info[0]->IsObject() && Nan::New(FileWrap::constructor)->HasInstance(info[0]->ToObject())) {
-                v8::Local<v8::Object> file_obj = info[0]->ToObject();
+            } else if (info[0]->IsObject() && Nan::New(FileWrap::constructor)->HasInstance(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
+                v8::Local<v8::Object> file_obj = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
                 BasicReaderWrap* reader_wrap = new BasicReaderWrap(unwrap<FileWrap>(file_obj), read_which_entities);
                 reader_wrap->Wrap(info.This());
                 info.GetReturnValue().Set(info.This());
@@ -83,16 +83,16 @@ namespace node_osmium {
         INSTANCE_CHECK(BasicReaderWrap, "BasicReader", "header");
         v8::Local<v8::Object> obj = Nan::New<v8::Object>();
         const osmium::io::Header& header = unwrap<BasicReaderWrap>(info.This()).header();
-        obj->Set(Nan::New(symbol_generator), Nan::New(header.get("generator")).ToLocalChecked());
+        Nan::Set(obj, Nan::New(symbol_generator), Nan::New(header.get("generator")).ToLocalChecked());
 
         auto bounds_array = Nan::New<v8::Array>(header.boxes().size());
 
         int i=0;
         for (const osmium::Box& box : header.boxes()) {
-            bounds_array->Set(i++, create_js_box(box));
+            Nan::Set(bounds_array, i++, create_js_box(box));
         }
 
-        obj->Set(Nan::New(symbol_bounds), bounds_array);
+        Nan::Set(obj, Nan::New(symbol_bounds), bounds_array);
         info.GetReturnValue().Set(obj);
         return;
     }

--- a/src/buffer_wrap.cpp
+++ b/src/buffer_wrap.cpp
@@ -28,7 +28,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "next", next);
         Nan::SetPrototypeMethod(lcons, "filter_point_in_time", filter_point_in_time);
         Nan::SetPrototypeMethod(lcons, "create_node_buffer", create_node_buffer);
-        target->Set(Nan::New(symbol_Buffer), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Buffer), lcons->GetFunction(target->CreationContext()).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -39,7 +39,7 @@ namespace node_osmium {
             info.GetReturnValue().Set(info.This());
             return;
         } else if (info.Length() == 1 && info[0]->IsObject()) {
-            auto obj = info[0]->ToObject();
+            auto obj = info[0]->ToObject(info.GetIsolate()->GetCurrentContext()).ToLocalChecked();
             if (node::Buffer::HasInstance(obj)) {
                 try {
                     osmium::memory::Buffer buffer(reinterpret_cast<unsigned char*>(node::Buffer::Data(obj)), node::Buffer::Length(obj));
@@ -49,7 +49,7 @@ namespace node_osmium {
                     // Store the node::Buffer in the new osmium::Buffer object
                     // so the node::Buffer doesn't go away if it goes out of scope
                     // outside this function.
-                    info.This()->Set(Nan::New("_data").ToLocalChecked(), obj);
+                    Nan::Set(info.This(), Nan::New("_data").ToLocalChecked(), obj);
                     info.GetReturnValue().Set(info.This());
                     return;
                 } catch (std::exception const& ex) {
@@ -76,7 +76,7 @@ namespace node_osmium {
 
         int filter_id = 0;
         if (info.Length() == 1 && info[0]->IsInt32()) {
-            filter_id = info[0]->Int32Value();
+            filter_id = Nan::To<int32_t>(info[0]).ToChecked();
         }
 
         while (buffer_wrap->m_iterator != buffer_wrap->m_this.end()) {
@@ -130,12 +130,12 @@ namespace node_osmium {
 
         osmium::Timestamp point_in_time;
         if (info[0]->IsInt32()) {
-            point_in_time = info[0]->Int32Value();
+            point_in_time = Nan::To<int32_t>(info[0]).ToChecked();
         } else if (info[0]->IsString()) {
             Nan::Utf8String time_string { info[0] };
             point_in_time = osmium::Timestamp(*time_string);
         } else if (info[0]->IsDate()) {
-            point_in_time = osmium::Timestamp(static_cast<int32_t>(v8::Date::Cast(*info[0])->NumberValue() / 1000));
+            point_in_time = osmium::Timestamp(static_cast<int32_t>(v8::Date::Cast(*info[0])->NumberValue(info.GetIsolate()->GetCurrentContext()).FromJust() / 1000));
         }
 
         typedef osmium::DiffIterator<osmium::memory::Buffer::t_iterator<osmium::OSMObject>> diff_iterator;
@@ -173,7 +173,7 @@ namespace node_osmium {
         v8::Local<v8::Object> global = Nan::GetCurrentContext()->Global();
 
         v8::Local<v8::Function> buffer_constructor =
-            v8::Local<v8::Function>::Cast(global->Get(Nan::New("Buffer").ToLocalChecked()));
+            v8::Local<v8::Function>::Cast(Nan::Get(global, Nan::New("Buffer").ToLocalChecked()).ToLocalChecked());
 
         v8::Local<v8::Value> constructor_info[3] = {
             slow_buffer,
@@ -183,7 +183,7 @@ namespace node_osmium {
 
         Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(buffer_constructor, 3, constructor_info);
         if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Buffer instance");
-        info.GetReturnValue().Set(maybe_local.ToLocalChecked()->ToObject());
+        info.GetReturnValue().Set(maybe_local.ToLocalChecked());
         return;
     }
 

--- a/src/file_wrap.cpp
+++ b/src/file_wrap.cpp
@@ -15,7 +15,7 @@ namespace node_osmium {
         v8::Local<v8::FunctionTemplate> lcons = Nan::New<v8::FunctionTemplate>(FileWrap::New);
         lcons->InstanceTemplate()->SetInternalFieldCount(1);
         lcons->SetClassName(Nan::New(symbol_File));
-        target->Set(Nan::New(symbol_File), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_File), lcons->GetFunction(target->CreationContext()).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -42,12 +42,11 @@ namespace node_osmium {
 
         try {
             osmium::io::File file;
-
             if (info[0]->IsString()) {
                 Nan::Utf8String filename { info[0] };
                 file = osmium::io::File(*filename, format);
-            } else if (info[0]->IsObject() && node::Buffer::HasInstance(info[0]->ToObject())) {
-                auto source = info[0]->ToObject();
+            } else if (info[0]->IsObject() && node::Buffer::HasInstance(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
+                auto source = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
                 file = osmium::io::File(node::Buffer::Data(source), node::Buffer::Length(source), format);
             } else {
                 Nan::ThrowTypeError(Nan::New("first argument to File constructor must be a string (filename) or node.Buffer").ToLocalChecked());

--- a/src/filter.cpp
+++ b/src/filter.cpp
@@ -20,7 +20,7 @@ namespace node_osmium {
 
     NAN_METHOD(Filter::register_filter) {
         if (info.Length() == 1 && info[0]->IsObject()) {
-            auto object = info[0]->ToObject();
+            auto object = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
             // XXX check that object is of class Filter
             Filter::all_filters.emplace_back(new Filter(object));
             assert(Filter::all_filters.size() < std::numeric_limits<int32_t>::max());
@@ -45,12 +45,12 @@ namespace node_osmium {
             m_entity_bits |= osmium::osm_entity_bits::from_item_type(item_type);
         }
         for (uint32_t i = 0; i < array->Length(); ++i) {
-            auto element = array->Get(i);
+            auto element = array->Get(Nan::GetCurrentContext(), i).ToLocalChecked();
             assert(element->IsArray());
             auto key_value = v8::Array::Cast(*element);
             assert(key_value->Length() == 2);
-            v8::Local<v8::Value> key = key_value->Get(0);
-            v8::Local<v8::Value> value = key_value->Get(1);
+            v8::Local<v8::Value> key = key_value->Get(Nan::GetCurrentContext(), 0).ToLocalChecked();
+            v8::Local<v8::Value> value = key_value->Get(Nan::GetCurrentContext(), 1).ToLocalChecked();
 
             if (key->IsString()) {
                 Nan::Utf8String key_string { key };
@@ -74,11 +74,11 @@ namespace node_osmium {
     Filter::Filter(v8::Local<v8::Object> object) :
         m_entity_bits(osmium::osm_entity_bits::nothing),
         m_tagged_entity_bits(osmium::osm_entity_bits::nothing) {
-        setup_filter(v8::Array::Cast(*object->Get(Nan::New("_node").ToLocalChecked())),      osmium::item_type::node);
-        setup_filter(v8::Array::Cast(*object->Get(Nan::New("_way").ToLocalChecked())),       osmium::item_type::way);
-        setup_filter(v8::Array::Cast(*object->Get(Nan::New("_relation").ToLocalChecked())),  osmium::item_type::relation);
-        setup_filter(v8::Array::Cast(*object->Get(Nan::New("_changeset").ToLocalChecked())), osmium::item_type::changeset);
-        setup_filter(v8::Array::Cast(*object->Get(Nan::New("_area").ToLocalChecked())),      osmium::item_type::area);
+        setup_filter(v8::Array::Cast(*object->Get(Nan::GetCurrentContext(), Nan::New("_node").ToLocalChecked()).ToLocalChecked()),      osmium::item_type::node);
+        setup_filter(v8::Array::Cast(*object->Get(Nan::GetCurrentContext(), Nan::New("_way").ToLocalChecked()).ToLocalChecked()),       osmium::item_type::way);
+        setup_filter(v8::Array::Cast(*object->Get(Nan::GetCurrentContext(), Nan::New("_relation").ToLocalChecked()).ToLocalChecked()),  osmium::item_type::relation);
+        setup_filter(v8::Array::Cast(*object->Get(Nan::GetCurrentContext(), Nan::New("_changeset").ToLocalChecked()).ToLocalChecked()), osmium::item_type::changeset);
+        setup_filter(v8::Array::Cast(*object->Get(Nan::GetCurrentContext(), Nan::New("_area").ToLocalChecked()).ToLocalChecked()),      osmium::item_type::area);
     }
 
     bool Filter::match(const osmium::OSMEntity& entity) const {

--- a/src/flex_reader_wrap.cpp
+++ b/src/flex_reader_wrap.cpp
@@ -29,7 +29,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "close", close);
         Nan::SetPrototypeMethod(lcons, "read", read);
         Nan::SetPrototypeMethod(lcons, "read_all", read_all);
-        target->Set(Nan::New(symbol_FlexReader), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_FlexReader), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -49,11 +49,11 @@ namespace node_osmium {
                     Nan::ThrowTypeError(Nan::New("Third argument to FlexReader constructor must be object").ToLocalChecked());
                     return;
                 }
-                read_which_entities = object_to_entity_bits(info[2]->ToObject());
+                read_which_entities = object_to_entity_bits(info[2]->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
             }
 
-            if (info[1]->IsObject() && Nan::New(LocationHandlerWrap::constructor)->HasInstance(info[1]->ToObject())) {
-                v8::Local<v8::Object> location_handler_obj = info[1]->ToObject();
+            if (info[1]->IsObject() && Nan::New(LocationHandlerWrap::constructor)->HasInstance(info[1]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
+                v8::Local<v8::Object> location_handler_obj = info[1]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
                 location_handler_type& location_handler = unwrap<LocationHandlerWrap>(location_handler_obj);
                 if (info[0]->IsString()) {
                     Nan::Utf8String filename { info[0] };
@@ -62,8 +62,8 @@ namespace node_osmium {
                     reader_wrap->Wrap(info.This());
                     info.GetReturnValue().Set(info.This());
                     return;
-                } else if (info[0]->IsObject() && Nan::New(FileWrap::constructor)->HasInstance(info[0]->ToObject())) {
-                    v8::Local<v8::Object> file_obj = info[0]->ToObject();
+                } else if (info[0]->IsObject() && Nan::New(FileWrap::constructor)->HasInstance(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
+                    v8::Local<v8::Object> file_obj = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
                     FlexReaderWrap* reader_wrap = new FlexReaderWrap(unwrap<FileWrap>(file_obj), location_handler, read_which_entities);
                     reader_wrap->Wrap(info.This());
                     info.GetReturnValue().Set(info.This());
@@ -88,16 +88,16 @@ namespace node_osmium {
         INSTANCE_CHECK(FlexReaderWrap, "FlexReader", "header");
         v8::Local<v8::Object> obj = Nan::New<v8::Object>();
         const osmium::io::Header& header = unwrap<FlexReaderWrap>(info.This()).header();
-        obj->Set(Nan::New(symbol_generator), Nan::New(header.get("generator")).ToLocalChecked());
+        Nan::Set(obj, Nan::New(symbol_generator), Nan::New(header.get("generator")).ToLocalChecked());
 
         auto bounds_array = Nan::New<v8::Array>(static_cast<unsigned int>(header.boxes().size()));
 
         int i=0;
         for (const osmium::Box& box : header.boxes()) {
-            bounds_array->Set(i++, create_js_box(box));
+            Nan::Set(bounds_array, i++, create_js_box(box));
         }
 
-        obj->Set(Nan::New(symbol_bounds), bounds_array);
+        Nan::Set(obj, Nan::New(symbol_bounds), bounds_array);
         info.GetReturnValue().Set(obj);
         return;
     }

--- a/src/handler.cpp
+++ b/src/handler.cpp
@@ -27,7 +27,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "on", on);
         Nan::SetPrototypeMethod(lcons, "options", options);
         Nan::SetPrototypeMethod(lcons, "end", stream_end);
-        target->Set(Nan::New(symbol_Handler), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Handler), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
 
     }
@@ -90,9 +90,9 @@ namespace node_osmium {
             return;
         }
 
-        v8::Local<v8::Value> tagged_nodes_only = info[0]->ToObject()->Get(Nan::New("tagged_nodes_only").ToLocalChecked());
+        v8::Local<v8::Value> tagged_nodes_only = info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked()->Get(Nan::GetCurrentContext(), Nan::New("tagged_nodes_only").ToLocalChecked()).ToLocalChecked();
         if (tagged_nodes_only->IsBoolean()) {
-            unwrap<JSHandler>(info.This()).node_callback_for_tagged_only = tagged_nodes_only->BooleanValue();
+            unwrap<JSHandler>(info.This()).node_callback_for_tagged_only = Nan::To<bool>(tagged_nodes_only).ToChecked();
         }
 
         info.GetReturnValue().Set(Nan::Undefined());
@@ -160,7 +160,7 @@ namespace node_osmium {
             return;
         }
         v8::Local<v8::Value> argv[1] = { new_external<TWrapped>(entity) };
-        Nan::New(function)->Call(Nan::GetCurrentContext()->Global(), 1, argv);
+        Nan::Call(Nan::New(function), Nan::GetCurrentContext()->Global(), 1, argv);
     }
 
     void call_callback(const Nan::Persistent<v8::Function>& function) {
@@ -168,7 +168,7 @@ namespace node_osmium {
         if (function.IsEmpty()) {
             return;
         }
-        Nan::New(function)->Call(Nan::GetCurrentContext()->Global(), 0, nullptr);
+        Nan::Call(Nan::New(function), Nan::GetCurrentContext()->Global(), 0, nullptr);
     }
 
     void JSHandler::dispatch_entity(const osmium::OSMEntity& entity) const {

--- a/src/location_handler_wrap.cpp
+++ b/src/location_handler_wrap.cpp
@@ -16,7 +16,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "clear", clear);
         Nan::SetPrototypeMethod(lcons, "ignoreErrors", ignoreErrors);
         Nan::SetPrototypeMethod(lcons, "end", stream_end);
-        target->Set(Nan::New(symbol_LocationHandler), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_LocationHandler), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 

--- a/src/multipolygon_collector_wrap.cpp
+++ b/src/multipolygon_collector_wrap.cpp
@@ -20,7 +20,7 @@ namespace node_osmium {
         lcons->SetClassName(Nan::New(symbol_MultipolygonCollector));
         Nan::SetPrototypeMethod(lcons, "read_relations", read_relations);
         Nan::SetPrototypeMethod(lcons, "handler", handler);
-        target->Set(Nan::New(symbol_MultipolygonCollector), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_MultipolygonCollector), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -56,11 +56,11 @@ namespace node_osmium {
             return;
         }
         try {
-            if (Nan::New(BasicReaderWrap::constructor)->HasInstance(info[0]->ToObject())) {
-                osmium::io::Reader& reader = unwrap<BasicReaderWrap>(info[0]->ToObject());
+            if (Nan::New(BasicReaderWrap::constructor)->HasInstance(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
+                osmium::io::Reader& reader = unwrap<BasicReaderWrap>(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
                 unwrap<MultipolygonCollectorWrap>(info.This()).read_relations(reader);
-            } else if (Nan::New(BufferWrap::constructor)->HasInstance(info[0]->ToObject())) {
-                osmium::memory::Buffer& buffer = unwrap<BufferWrap>(info[0]->ToObject());
+            } else if (Nan::New(BufferWrap::constructor)->HasInstance(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
+                osmium::memory::Buffer& buffer = unwrap<BufferWrap>(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
                 unwrap<MultipolygonCollectorWrap>(info.This()).read_relations(buffer.begin(), buffer.end());
             } else {
                 Nan::ThrowError(Nan::New("call MultipolygonCollector.read_relation() with BasicReader or Buffer object").ToLocalChecked());
@@ -78,12 +78,12 @@ namespace node_osmium {
 
     NAN_METHOD(MultipolygonCollectorWrap::handler) {
         INSTANCE_CHECK(MultipolygonCollectorWrap, "MultipolygonCollector", "handler");
-        if (info.Length() != 1 || !info[0]->IsObject() || !Nan::New(JSHandler::constructor)->HasInstance(info[0]->ToObject())) {
+        if (info.Length() != 1 || !info[0]->IsObject() || !Nan::New(JSHandler::constructor)->HasInstance(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked())) {
             Nan::ThrowError(Nan::New("call MultipolygonCollector.handler() with Handler object").ToLocalChecked());
             return;
         }
         try {
-            JSHandler& handler = unwrap<JSHandler>(info[0]->ToObject());
+            JSHandler& handler = unwrap<JSHandler>(info[0]->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
             auto& mc_handler = unwrap<MultipolygonCollectorWrap>(info.This()).handler([&handler](const osmium::memory::Buffer& area_buffer) {
                 for (const osmium::OSMEntity& entity : area_buffer) {
                     handler.dispatch_entity(entity);

--- a/src/multipolygon_handler_wrap.cpp
+++ b/src/multipolygon_handler_wrap.cpp
@@ -14,7 +14,7 @@ namespace node_osmium {
         lcons->InstanceTemplate()->SetInternalFieldCount(1);
         lcons->SetClassName(Nan::New(symbol_MultipolygonHandler));
         Nan::SetPrototypeMethod(lcons, "end", stream_end);
-        target->Set(Nan::New(symbol_MultipolygonHandler), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_MultipolygonHandler), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 

--- a/src/osm_area_wrap.cpp
+++ b/src/osm_area_wrap.cpp
@@ -26,7 +26,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "wkb", wkb);
         Nan::SetPrototypeMethod(lcons, "wkt", wkt);
         Nan::SetPrototypeMethod(lcons, "coordinates", coordinates);
-        target->Set(Nan::New(symbol_Area), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Area), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -84,9 +84,9 @@ namespace node_osmium {
         for (const auto& node_ref : node_ref_list) {
             const osmium::Location location = node_ref.location();
             v8::Local<v8::Array> coordinates = Nan::New<v8::Array>(2);
-            coordinates->Set(0, Nan::New(location.lon()));
-            coordinates->Set(1, Nan::New(location.lat()));
-            locations->Set(i, coordinates);
+            Nan::Set(coordinates, 0, Nan::New(location.lon()));
+            Nan::Set(coordinates, 1, Nan::New(location.lat()));
+            Nan::Set(locations, i, coordinates);
             ++i;
         }
         return locations;
@@ -95,7 +95,7 @@ namespace node_osmium {
     NAN_METHOD(OSMAreaWrap::coordinates) {
         INSTANCE_CHECK(OSMAreaWrap, "Area", "coordinates");
 
-        v8::Local<v8::Value> cf = Nan::New(module)->Get(Nan::New(symbol_Coordinates));
+        v8::Local<v8::Value> cf = Nan::Get(Nan::New(module), Nan::New(symbol_Coordinates)).ToLocalChecked();
         assert(cf->IsFunction());
 
         const osmium::Area& area = wrapped(info.This());
@@ -114,11 +114,11 @@ namespace node_osmium {
             unsigned array_size = 1u + inner_rings_range.size();
             v8::Local<v8::Array> ring = Nan::New<v8::Array>(array_size);
             int m = 0;
-            ring->Set(m++, get_coordinates(outer_ring));
+            Nan::Set(ring, m++, get_coordinates(outer_ring));
             for (const auto& inner_ring : inner_rings_range) {
-                ring->Set(m++, get_coordinates(inner_ring));
+                Nan::Set(ring, m++, get_coordinates(inner_ring));
             }
-            rings->Set(n, ring);
+            Nan::Set(rings, n, ring);
             ++n;
         }
 

--- a/src/osm_changeset_wrap.cpp
+++ b/src/osm_changeset_wrap.cpp
@@ -27,7 +27,7 @@ namespace node_osmium {
         ATTR(lcons, "closed", get_closed);
         ATTR(lcons, "bounds", get_bounds);
         Nan::SetPrototypeMethod(lcons, "tags", tags);
-        target->Set(Nan::New(symbol_Changeset), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Changeset), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 

--- a/src/osm_entity_wrap.cpp
+++ b/src/osm_entity_wrap.cpp
@@ -12,7 +12,7 @@ namespace node_osmium {
         v8::Local<v8::FunctionTemplate> lcons = Nan::New<v8::FunctionTemplate>(OSMEntityWrap::New);
         lcons->InstanceTemplate()->SetInternalFieldCount(1);
         lcons->SetClassName(Nan::New(symbol_OSMEntity));
-        target->Set(Nan::New(symbol_OSMEntity), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_OSMEntity), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 

--- a/src/osm_entity_wrap.hpp
+++ b/src/osm_entity_wrap.hpp
@@ -30,7 +30,7 @@ namespace node_osmium {
                 case 0: {
                     v8::Local<v8::Object> tags = Nan::New<v8::Object>();
                     for (const auto& tag : object.tags()) {
-                        tags->Set(Nan::New(tag.key()).ToLocalChecked(), Nan::New(tag.value()).ToLocalChecked());
+                        Nan::Set(tags, Nan::New(tag.key()).ToLocalChecked(), Nan::New(tag.value()).ToLocalChecked());
                     }
                     info.GetReturnValue().Set(tags);
                     return;

--- a/src/osm_node_wrap.cpp
+++ b/src/osm_node_wrap.cpp
@@ -27,7 +27,7 @@ namespace node_osmium {
         ATTR(lcons, "coordinates", get_coordinates);
         ATTR(lcons, "lon", get_lon);
         ATTR(lcons, "lat", get_lat);
-        target->Set(Nan::New(symbol_Node), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Node), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -44,14 +44,14 @@ namespace node_osmium {
     }
 
     NAN_GETTER(OSMNodeWrap::get_coordinates) {
-        auto cf = Nan::New(module)->Get(Nan::New(symbol_Coordinates));
+        auto cf = Nan::Get(Nan::New(module), Nan::New(symbol_Coordinates)).ToLocalChecked();
         assert(cf->IsFunction());
 
         const osmium::Location& location = wrapped(info.This()).location();
         if (!location) {
             Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(v8::Local<v8::Function>::Cast(cf));
             if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Buffer instance");
-            info.GetReturnValue().Set(maybe_local.ToLocalChecked()->ToObject());
+            info.GetReturnValue().Set(maybe_local.ToLocalChecked());
             return;
         }
 
@@ -60,7 +60,7 @@ namespace node_osmium {
         v8::Local<v8::Value> argv[2] = { lon, lat };
         Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(v8::Local<v8::Function>::Cast(cf),2,argv);
         if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Buffer instance");
-        info.GetReturnValue().Set(maybe_local.ToLocalChecked()->ToObject());
+        info.GetReturnValue().Set(maybe_local.ToLocalChecked());
         return;
     }
 

--- a/src/osm_object_wrap.cpp
+++ b/src/osm_object_wrap.cpp
@@ -20,7 +20,7 @@ namespace node_osmium {
         ATTR(lcons, "uid", get_uid);
         ATTR(lcons, "user", get_user);
         Nan::SetPrototypeMethod(lcons, "tags", tags);
-        target->Set(Nan::New(symbol_OSMObject), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_OSMObject), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 

--- a/src/osm_relation_wrap.cpp
+++ b/src/osm_relation_wrap.cpp
@@ -15,7 +15,7 @@ namespace node_osmium {
         ATTR(lcons, "type", get_type);
         ATTR(lcons, "members_count", get_members_count);
         Nan::SetPrototypeMethod(lcons, "members", members);
-        target->Set(Nan::New(symbol_Relation), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Relation), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -47,10 +47,10 @@ namespace node_osmium {
                 for (const auto& member : relation.members()) {
                     v8::Local<v8::Object> jsmember = Nan::New<v8::Object>();
                     typec[0] = osmium::item_type_to_char(member.type());
-                    jsmember->Set(Nan::New(symbol_type), Nan::New(typec).ToLocalChecked());
-                    jsmember->Set(Nan::New(symbol_ref), Nan::New<v8::Number>(member.ref()));
-                    jsmember->Set(Nan::New(symbol_role), Nan::New(member.role()).ToLocalChecked());
-                    members->Set(i, jsmember);
+                    Nan::Set(jsmember, Nan::New(symbol_type), Nan::New(typec).ToLocalChecked());
+                    Nan::Set(jsmember, Nan::New(symbol_ref), Nan::New<v8::Number>(member.ref()));
+                    Nan::Set(jsmember, Nan::New(symbol_role), Nan::New(member.role()).ToLocalChecked());
+                    Nan::Set(members, i, jsmember);
                     ++i;
                 }
                 info.GetReturnValue().Set(members);
@@ -61,7 +61,7 @@ namespace node_osmium {
                     Nan::ThrowTypeError(Nan::New("call members() without parameters or the index of the member you want").ToLocalChecked());
                     return;
                 }
-                uint32_t n = info[0]->Uint32Value();;
+                uint32_t n = Nan::To<uint32_t>(info[0]).ToChecked();
                 if (n < relation.members().size()) {
                     auto it = relation.members().begin();
                     std::advance(it, n);
@@ -69,9 +69,9 @@ namespace node_osmium {
                     v8::Local<v8::Object> jsmember = Nan::New<v8::Object>();
                     char typec[2] = " ";
                     typec[0] = osmium::item_type_to_char(member.type());
-                    jsmember->Set(Nan::New(symbol_type), Nan::New(typec).ToLocalChecked());
-                    jsmember->Set(Nan::New(symbol_ref), Nan::New<v8::Number>(member.ref()));
-                    jsmember->Set(Nan::New(symbol_role), Nan::New(member.role()).ToLocalChecked());
+                    Nan::Set(jsmember, Nan::New(symbol_type), Nan::New(typec).ToLocalChecked());
+                    Nan::Set(jsmember, Nan::New(symbol_ref), Nan::New<v8::Number>(member.ref()));
+                    Nan::Set(jsmember, Nan::New(symbol_role), Nan::New(member.role()).ToLocalChecked());
                     info.GetReturnValue().Set(jsmember);
                     return;
                 } else {

--- a/src/osm_way_wrap.cpp
+++ b/src/osm_way_wrap.cpp
@@ -26,7 +26,7 @@ namespace node_osmium {
         Nan::SetPrototypeMethod(lcons, "node_coordinates", node_coordinates);
         Nan::SetPrototypeMethod(lcons, "wkb", wkb);
         Nan::SetPrototypeMethod(lcons, "wkt", wkt);
-        target->Set(Nan::New(symbol_Way), lcons->GetFunction());
+        Nan::Set(target, Nan::New(symbol_Way), Nan::GetFunction(lcons).ToLocalChecked());
         constructor.Reset(lcons);
     }
 
@@ -80,7 +80,7 @@ namespace node_osmium {
                 v8::Local<v8::Array> nodes = Nan::New<v8::Array>(way.nodes().size());
                 int i = 0;
                 for (const auto& node_ref : way.nodes()) {
-                    nodes->Set(i, Nan::New<v8::Number>(node_ref.ref()));
+                    Nan::Set(nodes, i, Nan::New<v8::Number>(node_ref.ref()));
                     ++i;
                 }
                 info.GetReturnValue().Set(nodes);
@@ -91,7 +91,7 @@ namespace node_osmium {
                     Nan::ThrowTypeError(Nan::New("call node_refs() without parameters or the index of the node you want").ToLocalChecked());
                     return;
                 }
-                uint32_t n = info[0]->Uint32Value();
+                uint32_t n = Nan::To<uint32_t>(info[0]).ToChecked();
                 if (n < way.nodes().size()) {
                     info.GetReturnValue().Set(Nan::New<v8::Number>(way.nodes()[n].ref()));
                     return;
@@ -108,7 +108,7 @@ namespace node_osmium {
 
     NAN_METHOD(OSMWayWrap::node_coordinates) {
 
-        auto cf = Nan::New(module)->Get(Nan::New(symbol_Coordinates));
+        auto cf = Nan::Get(Nan::New(module), Nan::New(symbol_Coordinates)).ToLocalChecked();
         assert(cf->IsFunction());
 
         const osmium::Way& way = wrapped(info.This());
@@ -130,7 +130,7 @@ namespace node_osmium {
                             v8::Local<v8::Value> argv[2] = { Nan::New(location.lon()), Nan::New<v8::Number>(location.lat()) };
                             Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(v8::Local<v8::Function>::Cast(cf), 2, argv);
                             if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Node instance");
-                            nodes->Set(i, maybe_local.ToLocalChecked()->ToObject());
+                            Nan::Set(nodes, i, maybe_local.ToLocalChecked());
                             ++i;
                             last_location = location;
                         }
@@ -151,14 +151,14 @@ namespace node_osmium {
                     Nan::ThrowTypeError(Nan::New("call node_coordinates() without parameters or the index of the node you want").ToLocalChecked());
                     return;
                 }
-                uint32_t n = info[0]->Uint32Value();
+                uint32_t n = Nan::To<uint32_t>(info[0]).ToChecked();
                 if (n < way.nodes().size()) {
                     const osmium::Location location = way.nodes()[n].location();
                     if (location.valid()) {
                         v8::Local<v8::Value> argv[2] = { Nan::New(location.lon()), Nan::New<v8::Number>(location.lat()) };
                         Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(v8::Local<v8::Function>::Cast(cf), 2, argv);
                         if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Node instance");
-                        info.GetReturnValue().Set(maybe_local.ToLocalChecked()->ToObject());
+                        info.GetReturnValue().Set(maybe_local.ToLocalChecked());
                         return;
                     } else {
                         info.GetReturnValue().Set(Nan::Undefined());

--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -16,55 +16,55 @@ namespace node_osmium {
             return scope.Escape(Nan::Undefined());
         }
 
-        auto cf = Nan::New(module)->Get(Nan::New(symbol_Coordinates));
+        auto cf = Nan::Get(Nan::New(module), Nan::New(symbol_Coordinates)).ToLocalChecked();
         assert(cf->IsFunction());
-        auto bf = Nan::New(module)->Get(Nan::New(symbol_Box));
+        auto bf = Nan::Get(Nan::New(module), Nan::New(symbol_Box)).ToLocalChecked();
         assert(bf->IsFunction());
 
         v8::Local<v8::Value> argv_bl[2] = { Nan::New(box.bottom_left().lon()), Nan::New<v8::Number>(box.bottom_left().lat()) };
 
         Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(v8::Local<v8::Function>::Cast(cf),2,argv_bl);
         if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Box instance");
-        auto bottom_left = maybe_local.ToLocalChecked()->ToObject();
+        auto bottom_left = maybe_local.ToLocalChecked()->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
 
         v8::Local<v8::Value> argv_tr[2] = { Nan::New(box.top_right().lon()), Nan::New<v8::Number>(box.top_right().lat()) };
         Nan::MaybeLocal<v8::Object> maybe_local2 = Nan::NewInstance(v8::Local<v8::Function>::Cast(cf),2,argv_tr);
         if (maybe_local2.IsEmpty()) Nan::ThrowError("Could not create new Box instance");
-        auto top_right = maybe_local2.ToLocalChecked()->ToObject();
+        auto top_right = maybe_local2.ToLocalChecked()->ToObject(Nan::GetCurrentContext()).ToLocalChecked();
 
         v8::Local<v8::Value> argv_box[2] = { bottom_left, top_right };
 
         Nan::MaybeLocal<v8::Object> maybe_local3 = Nan::NewInstance(v8::Local<v8::Function>::Cast(bf),2,argv_box);
         if (maybe_local3.IsEmpty()) Nan::ThrowError("Could not create new Box instance");
 
-        return scope.Escape(maybe_local3.ToLocalChecked()->ToObject());
+        return scope.Escape(maybe_local3.ToLocalChecked()->ToObject(Nan::GetCurrentContext()).ToLocalChecked());
     }
 
     osmium::osm_entity_bits::type object_to_entity_bits(v8::Local<v8::Object> options) {
         osmium::osm_entity_bits::type entities = osmium::osm_entity_bits::nothing;
 
-        v8::Local<v8::Value> want_nodes = options->Get(Nan::New(symbol_node));
-        if (want_nodes->IsBoolean() && want_nodes->BooleanValue()) {
+        v8::Local<v8::Value> want_nodes = Nan::Get(options, Nan::New(symbol_node)).ToLocalChecked();
+        if (want_nodes->IsBoolean() && Nan::To<bool>(want_nodes).ToChecked()) {
             entities |= osmium::osm_entity_bits::node;
         }
 
-        v8::Local<v8::Value> want_ways = options->Get(Nan::New(symbol_way));
-        if (want_ways->IsBoolean() && want_ways->BooleanValue()) {
+        v8::Local<v8::Value> want_ways = Nan::Get(options, Nan::New(symbol_way)).ToLocalChecked();
+        if (want_ways->IsBoolean() && Nan::To<bool>(want_ways).ToChecked()) {
             entities |= osmium::osm_entity_bits::way;
         }
 
-        v8::Local<v8::Value> want_relations = options->Get(Nan::New(symbol_relation));
-        if (want_relations->IsBoolean() && want_relations->BooleanValue()) {
+        v8::Local<v8::Value> want_relations = Nan::Get(options, Nan::New(symbol_relation)).ToLocalChecked();
+        if (want_relations->IsBoolean() && Nan::To<bool>(want_relations).ToChecked()) {
             entities |= osmium::osm_entity_bits::relation;
         }
 
-        v8::Local<v8::Value> want_areas = options->Get(Nan::New(symbol_area));
-        if (want_areas->IsBoolean() && want_areas->BooleanValue()) {
+        v8::Local<v8::Value> want_areas = Nan::Get(options, Nan::New(symbol_area)).ToLocalChecked();
+        if (want_areas->IsBoolean() && Nan::To<bool>(want_areas).ToChecked()) {
             entities |= osmium::osm_entity_bits::area;
         }
 
-        v8::Local<v8::Value> want_changesets = options->Get(Nan::New(symbol_changeset));
-        if (want_changesets->IsBoolean() && want_changesets->BooleanValue()) {
+        v8::Local<v8::Value> want_changesets = Nan::Get(options, Nan::New(symbol_changeset)).ToLocalChecked();
+        if (want_changesets->IsBoolean() && Nan::To<bool>(want_changesets).ToChecked()) {
             entities |= osmium::osm_entity_bits::changeset;
         }
 

--- a/src/utils.hpp
+++ b/src/utils.hpp
@@ -35,10 +35,11 @@ namespace node_osmium {
     template<class T, class... Args>
     v8::Local<v8::Object> new_external(Args&&... args) {
         Nan::EscapableHandleScope scope;
-        v8::Handle<v8::Value> ext = Nan::New<v8::External>(new T(std::forward<Args>(args)...));
-        Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(Nan::New(T::constructor)->GetFunction(), 1, &ext);
+        v8::Local<v8::Value> ext = Nan::New<v8::External>(new T(std::forward<Args>(args)...));
+        v8::Local<v8::Function> f = Nan::GetFunction(Nan::New(T::constructor)).ToLocalChecked();
+        Nan::MaybeLocal<v8::Object> maybe_local = Nan::NewInstance(f, 1, &ext);
         if (maybe_local.IsEmpty()) Nan::ThrowError("Could not create new Buffer instance");
-        return scope.Escape(maybe_local.ToLocalChecked()->ToObject());
+        return scope.Escape(maybe_local.ToLocalChecked());
 
     }
 

